src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -931,4 +938,62 @@ public Response getHasBeenDeleted(@Context ContainerRequestContext crc, @PathPar
             return ok(dataFileServiceBean.hasBeenDeleted(dataFile));
         }, getRequestUser(crc));
     }
+
+    /**
+     * @param fileIdOrPersistentId Database ID or PID of the data file.
+     * @param dsVersionString The version of the dataset, such as 1.0, :draft,
+     * :latest-published, etc.
+     */
+    @GET
+    @AuthRequired
+    @Path("{id}/versions/{dsVersionString}/citation")
+    public Response getFileCitationByVersion(@Context ContainerRequestContext crc, @PathParam("id") String fileIdOrPersistentId, @PathParam("dsVersionString") String dsVersionString) {
+        try {
+            DataverseRequest req = createDataverseRequest(getRequestUser(crc));
+            final DataFile df = execCommand(new GetDataFileCommand(req, findDataFileOrDie(fileIdOrPersistentId)));
+            Dataset ds = df.getOwner();
+            // Adapted from getDatasetVersionOrDie
+            DatasetVersion dsv = execCommand(handleVersion(dsVersionString, new Datasets.DsVersionHandler<Command<DatasetVersion>>() {
+
+                boolean includeDeaccessioned = true;
```

Comment:
To ensure consistency across endpoints and make the API independent of the SPA, I suggest introducing an optional query parameter, includeDeaccessioned, with a default value set to false, as we do for other endpoints.

Reply:
Address in this change: 4542b21310 refactor handleVersion, add includeDeaccession param